src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -62,139 +53,175 @@ export interface MenuProps {
+    setNegativeTabindexForNonInitialMenuItems();
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    if (props.enableKeyboardNavigation
+        && usedMenuTemplateRefs.value.length < 1
+    ) {
+      console.warn('@infermedica/component-library: use itemsTemplateRefs to pass UiMenuItems template refs.');
+    }
+  }
+});
+const lastFocusedMenuItemTemplateRefs = ref<ElementRef | null>(null);
+const handleMenuItemFocus = async (element: ElementRef | InstanceType<typeof UiMenuItem> | undefined) => {
+  if (element && initialElement.value) {
+    await focusElement(element?.itemTemplateRefs?.content?.$el, true);
+    initialElement.value.tabindex = -1;
+  }
+};
+const handleMenuKeyDown = async ({ key }: KeyboardEvent) => {
+  if (!props.enableKeyboardNavigation) {
+    return;
+  }
+  const activeElement = focusedElement.value;
   switch (key) {
     case 'ArrowUp':
-      if (prevMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        prevMenuItem.value.tabindex = 0;
-        focusElement(prevMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(prevElement.value);
       break;
     case 'ArrowDown':
-      if (nextMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        nextMenuItem.value.tabindex = 0;
-        focusElement(nextMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(nextElement.value);
       break;
     case 'Home':
     case 'PageUp':
-      if (firstMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        firstMenuItem.value.tabindex = 0;
-        focusElement(firstMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(usedMenuTemplateRefs.value.at(0));
       break;
     case 'End':
     case 'PageDown':
-      if (lastMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        lastMenuItem.value.tabindex = 0;
-        focusElement(lastMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(usedMenuTemplateRefs.value.at(-1));
       break;
     case 'Tab':
-      const lastFocusedElement = ref(focusedMenuItem.value);
-      setTimeout(() => {
-        if(!lastFocusedElement.value) return;
-        lastFocusedElement.value.tabindex = -1;
-        if (selectedMenuItem.value) {
-          selectedMenuItem.value.tabindex = 0;
-          return;
-        }
-        if (firstMenuItem.value) {
-          firstMenuItem.value.tabindex = 0;
-        }
-      }, 0);
+      if (initialElement.value === activeElement) {
+        return;
+      }
+      if (activeElement) {
+        activeElement.tabindex = 0;
+        // We need to use setTimeout() because nextTick() does not bring the desired effect.
+        window.setTimeout(() => {
+          activeElement.tabindex = -1;
+        }, 0);
+      }
+      break;
```

Comment:
And another comment about indentations 😅  Doesn't it look neater? 😄 
```suggestion
      if (!activeElement) break;
      activeElement.tabindex = 0;
      // We need to use setTimeout() because nextTick() does not bring the desired effect.
      window.setTimeout(() => {
        activeElement.tabindex = -1;
      }, 0);
      break;
```

Reply:
TBH it depends, because when I use collapsed files 🙈 my pattern looks "neater" :D 
![Zrzut ekranu 2024-05-14 o 10 13 27](https://github.com/infermedica/component-library/assets/12138170/c1b0c7b1-7adb-4655-9975-82b5a1800c35)

Reply:
Well... That's an argument 🤔 😎 tbh, I barely use _collapsing_ in the code and its' fragments as I treat most of the code with similar importance. But maybe it's not right, perhaps I shouldn't, and maybe I should start using collapsing 🤔  I'm confused right now, both syntaxes make sense to me right now 😅 😓